text_to_image/stable_diffusion/run.py

```diff
@@ -0,0 +1,95 @@
+import os
+import sys
+
+import torch
+import pathlib
+from pathlib import Path
+from utils.downloads.utils import get_downloads_path
+
+
+def run_pytorch_fp32(model_path, config, steps, scale, prompt, batch_size, num_runs, timeout):
+    sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "stablediffusion"))
+
+    from omegaconf import OmegaConf
+    from contextlib import nullcontext
+    from utils.benchmark import run_model
+    from utils.pytorch import PyTorchRunnerV2
+    from pytorch_lightning import seed_everything
+    from utils.text_to_image.stable_diffusion import StableDiffusion
+    from text_to_image.stable_diffusion.stablediffusion.ldm.models.diffusion.ddim import DDIMSampler
+    from text_to_image.stable_diffusion.stablediffusion.scripts.txt2img import load_model_from_config
+
+    seed_everything(42)
+    config = OmegaConf.load(f"{config}")
+    model = load_model_from_config(config, f"{model_path}", torch.device("cpu"))
+    sampler = DDIMSampler(model, device=torch.device("cpu"))
+    shape = [4, 512 // 8, 512 // 8]
+
+    # TODO: torchscript stuff, should it stay here?
+    unet = model.model.diffusion_model
+    decoder = model.first_stage_model.decoder
+
+    stablediffusion_data = pathlib.Path(get_downloads_path(), "stable_diffusion")
+    unet_path = Path(stablediffusion_data, "unet.pt")
+    decoder_path = Path(stablediffusion_data, "decoder.pt")
+    if not stablediffusion_data.exists():
+        stablediffusion_data.mkdir(exist_ok=True)
+
+    with torch.no_grad(), nullcontext():
+        if unet_path.exists():
+            scripted_unet = torch.jit.load(unet_path)
+        else:
+            img_in = torch.ones(2, 4, 96, 96, dtype=torch.float32)
+            t_in = torch.ones(2, dtype=torch.int64)
+            context = torch.ones(2, 77, 1024, dtype=torch.float32)
+            scripted_unet = torch.jit.trace(unet, (img_in, t_in, context))
+            scripted_unet = torch.jit.freeze(scripted_unet)
+            torch.jit.save(scripted_unet, unet_path)
+        model.model.scripted_diffusion_model = scripted_unet
+
+        if decoder_path.exists():
+            scripted_decoder = torch.jit.load(decoder_path)
+        else:
+            samples_ddim = torch.ones(1, 4, 96, 96, dtype=torch.float32)
+            scripted_decoder = torch.jit.trace(decoder, (samples_ddim))
+            scripted_decoder = torch.jit.freeze(scripted_decoder)
+            torch.jit.save(scripted_decoder, decoder_path)
+        model.first_stage_model.decoder = scripted_decoder
+
+    def single_pass_pytorch(_runner, _stablediffusion):
+        _runner.run(batch_size * steps)
+        _stablediffusion.submit_count()
```

Comment:
it would be better to submit count based on batch size to know how many images have been generated (even though we dont measure accuracy yet)